readthedocs/api/v3/urls.py

```diff
@@ -24,6 +24,7 @@
 # allows /api/v3/projects/
 # allows /api/v3/projects/pip/
 # allows /api/v3/projects/pip/superproject/
+# allows /api/v3/projects/pip/sync_versions/
```

Comment:
I would use `PATCH /api/v3/projects/<project_slug>/versions/` here instead a POST on a new URL `sync_versions`. This is to keep consistency with how the API is structured.

Reply:
I like having a specific endpoint. PATCH is meant to update a resource, having a PATCH over /versions isn't clear what it does, a specific endpoint is more explicit, if we have other operations over versions we won't be able to use another verb here. And having another endpoint makes it easier to define the permissions needed for that operation.

Reply:
My reasoning behind this was to follow the semantic meaning and pattern we already have with other endpoints:

* GET to get resources (detail or list)
* POST to create a new resource
* PATCH to update the resources

Note we can perform multiple HTTP verbs on the same endpoint: for example GET over `/builds` will return a list of builds and POST over `/builds` will create a new build. This is already implemented.

I applied the same logic here where GET over `/versions` will return a list of versions and PATCH on `/versions` will _update all the versions_ we have in the database with the branches/tags on the Git repository. The "sync" operation is performed on the `Version` resources, is not performed on the `Project`; and that's why it has to be under `/versions` in my opinion.

----

That said, it seems that @ericholscher found this proposal is hard to implement (based on a comment on Slack), so I'm fine if you prefer to follow a different approach here. I will remove "Request changes" revision.

Reply:
I agree with @stsewd here. I think that `PATCH /projects/$project/versions/` could do a number of different things. Say I wanted to update a field across all versions, it seems like that's also what PATCH to that endpoint would do? It definitely isn't obvious to me that it would cause a background task to sync things, instead of allowing me to explicitly set data across the whole endpoint.

I'm happy to move the endpoint to something like `POST /projects/$project/versions/sync/` if we really want it to live on `/verisons/`, but I also view this as an action on the project itself, but could be convinced otherwise.

Reply:
> I'm happy to move the endpoint to something like POST /projects/$project/versions/sync/ if we really want it to live on /verisons/, but I also view this as an action on the project itself, but could be convinced otherwise.

We can't use `projects/{project}/versions/sync`, as it will collide with a version named `sync`.